components/p3a/brave_p3a_service.cc

```diff
@@ -194,8 +194,10 @@ base::TimeDelta TimeDeltaTillMonday(base::Time time) {
     days_till_monday = 1;
   }
 
-  base::TimeDelta result = base::TimeDelta::FromDays(days_till_monday) -
-                           (time - time.LocalMidnight());
+  // Adding few hours of padding to prevent potential problems with DST.
+  base::Time result =
+      (time.LocalMidnight() + base::Days(days_till_monday) + base::Hours(4))
+          .LocalMidnight();
   return result;
```

Comment:
tbh, I'd prefer to use UTC and just add a random 24 hours offset on top. I don't see any reason to use Local time here. It might be even better from a server POV.

a proper random time delta can be taken from here: https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/metrics/perf/metric_collector.cc;l=31-35?q=RandomTimeDelta&ss=chromium

Reply:
I also prefer UTC, but unfortunately it was decided to use local time for consistency with the usage pings and also to decrease load on brave servers. Random offset is also not part of the spec. So I'd leave this question for the potential follow-ups.